### PR TITLE
DTSPO-441:: Users offboarding

### PR DESCRIPTION
**Pull request by mokainos**
https://tools.hmcts.net/jira/browse/DTSPO-441

### CODE REVIEW

File: azure-pipelines-launchdarkly-offboard-inactive.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: offboard-launchdarkly-user
+
+pr:
+  paths:
+    include:
+      - azure-pipelines-launchdarkly-offboard-inactive.yml
+
+schedules:
+- cron: "0 12 * * 0"
+  displayName:  Weekly Sunday
```
Comment (timja) [inline azure-pipelines-launchdarkly-offboard-inactive.yml:10]:
```suggestion
  displayName: Weekly Sunday
```

File: azure-pipelines-launchdarkly-offboard-inactive.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  paths:
+    include:
+      - azure-pipelines-launchdarkly-offboard-inactive.yml
+
+schedules:
+- cron: "0 12 * * 0"
+  displayName:  Weekly Sunday 
+  branches:
+    include:
+    - main
```
Comment (timja) [inline azure-pipelines-launchdarkly-offboard-inactive.yml:13]:
```suggestion
    - master
```

File: azure-pipelines-launchdarkly-offboard.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: offboard-launchdarkely-user
+
+pr:
+  paths:
+    include:
+      - azure-pipelines-launchdarkely-offboard.yml
+
+schedules:
+- cron: "0 12 * * 0"
+  displayName: cd Weekly Sunday
```
[the remark this answers is not in the document]
Comment (mokainos) [reply azure-pipelines-launchdarkly-offboard.yml]: modified

File: azure-pipelines-launchdarkly-offboard-inactive.yml (hunk 2 above)
Comment (mokainos) [reply azure-pipelines-launchdarkly-offboard-inactive.yml]: modified

File: pipeline-scripts/launchdarkly-offboard.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
+shopt -s nocasematch
+window=5097615
+currenttime=$(date +%s)
+IFS=,
+
+function deleteuser() {
```
Comment (timja) [inline pipeline-scripts/launchdarkly-offboard.sh:8]: this doesn't appear to be called?

Comment (mokainos) [reply pipeline-scripts/launchdarkly-offboard.sh:8]: deleteuser $id
USED IN LINE 21/27/36

Comment (timja) [reply pipeline-scripts/launchdarkly-offboard.sh:8]: o right missed the closing brace

File: pipeline-scripts/launchdarkly-offboard.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          --header 'LD-API-Version: beta'
+
+}
+curl -s --request GET   --url 'https://app.launchdarkly.com/api/v2/members?limit=200&offset=0'   --header "Authorization: $TOKEN"   --header 'LD-API-Version: beta' | jq -r '.items | map({_id, email, _lastSeen})| (first | keys_unsorted) as $keys | map([to_entries[] | .value]) as $rows | $keys,$rows[] | @csv' | tr -d \" | grep -v _lastSeen| while read id email lastseen
+do
+   # delete an user based on argument passed
+    if [[ $email == "$EMAIL" ]]
+    then
+      echo "Deleting user $EMAIL"
+      deleteuser $id
```
Comment (timja) [inline pipeline-scripts/launchdarkly-offboard.sh:21]: I think we can just exit from here?
```suggestion
      deleteuser $id
      exit 0
```

**Review by timja — APPROVED** [review]

**Pull request merged**